misc/watcher.py

```diff
@@ -41,23 +49,49 @@ def execute_ocrmypdf(file_path):
         output_path = Path(output_directory_year_month) / filename
     else:
         output_path = Path(OUTPUT_DIRECTORY) / filename
-    print(f'New file: {file_path}.\nAttempting to OCRmyPDF to: {output_path}')
-    ocrmypdf.ocr(file_path, output_path)
+    logger.info(f'New file: {file_path}. Waiting until fully loaded...')
+    # This loop waits to make sure that the file is completely loaded on
+    # disk before attempting to read. Docker sometimes will publish the
+    # watchdog event before the file is actually fully on disk, causing
+    # pikepdf to fail.
+    current_size = None
+    while current_size != new_file.stat().st_size:
```

Comment:
This is defeated by opening a Python terminal and running
`f = open('file.pdf', 'wb')` in the input directory. In other words if a program open a file and has any sizable wait before it begins writing then, we'll trigger on a file size of zero. Or during writing a delay of more than 1 second is entirely possible on a slow network. In some file systems even opening a folder with >1000 files can see this kind of delay.

This is really something watchdog should figure out for us....

Reply:
Also consider the following:
* Samba client opens new file for writing.
* Samba client writes some bytes.
* Watcher enters loop waiting for file size to change.
* Samba client crashes/goes offline without finishing.
* Watcher is stuck indefinitely waiting for that one file, even as other files are changed.